src/endpoint/s3/s3_utils.js

```diff
@@ -277,6 +277,7 @@ function set_response_object_md(res, object_md) {
     if (object_md.version_id) res.setHeader('x-amz-version-id', object_md.version_id);
     set_response_xattr(res, object_md.xattr);
     if (object_md.tag_count) res.setHeader('x-amz-tagging-count', object_md.tag_count);
+    if (object_md.part_count) res.setHeader('x-amz-mp-parts-count', object_md.part_count);
```

Comment:
I don't want to confuse our internal structure between our own object parts count and the multipart count so I suggest to rename this `object_md.multipart_count`